src/popup/components/TransactionOverview.vue

```diff
@@ -5,106 +5,125 @@
     :sender="transaction.sender"
     :recipient="transaction.recipient"
     :tx-function="transaction.function"
+    :tx="tx"
   />
 </template>
 
-<script>
-import { mapState, mapGetters } from 'vuex';
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  ref,
+  onMounted,
+} from '@vue/composition-api';
 import { SCHEMA } from '@aeternity/aepp-sdk';
+import { useGetter } from '../../composables';
 import { watchUntilTruthy } from '../utils';
+import { IAccount } from '../../types';
+
 import TransactionInfo from './TransactionInfo.vue';
 
-export default {
+export default defineComponent({
+  name: 'TransactionOverview',
   components: { TransactionInfo },
   props: {
     tx: { type: Object, required: true },
     isDex: { type: Boolean, default: false },
   },
-  data: () => ({ name: '' }),
-  computed: {
-    ...mapGetters(['getTxType', 'getTxDirection', 'getExplorerPath', 'getDexContracts']),
-    ...mapGetters('names', ['getPreferred']),
-    ...mapState({
-      account(_, { account }) {
-        return {
-          ...account,
-          label: this.$t('transaction.overview.accountAddress'),
-          url: this.getExplorerPath(account.address),
-        };
-      },
-      isDexRecipient() {
-        return [...this.getDexContracts.router, ...this.getDexContracts.wae]
-          .includes(this.tx?.contractId);
-      },
-    }),
-    transaction() {
-      switch (this.txType) {
+  setup(props, { root }) {
+    const name = ref<string>();
+    const getTxType = useGetter('getTxType');
+    const getTxDirection = useGetter('getTxDirection');
+    const getExplorerPath = useGetter('getExplorerPath');
+    const getDexContracts = useGetter('getDexContracts');
+    const getPreferred = computed(() => root.$store.getters['names/getPreferred']);
+    const _account = useGetter<IAccount>('account');
+    const account: any = computed<any>(() => ({
```

Comment:
I suggest to avoid using `any` to type things that can be typed. Also if this computed is not an `IAccount` I suggest to rename it to something like `parsedAccount` or `computedAccount` or `accountFor...`.